code/datums/components/stun_mitigation.dm

```diff
@@ -0,0 +1,186 @@
+				slot = SLOT_WEAR_ID
+			else if(parent == holder_human.r_store)
+				slot = SLOT_R_STORE
+			else if(parent == holder_human.l_store)
+				slot = SLOT_L_STORE
+			else if(parent == holder_human.s_store)
+				slot = SLOT_S_STORE
+	shield_equipped(parent, holder_mob, slot)
+
+///Toggles the mitigation on or off when already equipped
+/datum/component/stun_mitigation/proc/toggle_shield/(datum/source, new_state)
+	SIGNAL_HANDLER
+	if(active == new_state)
+		return
+
+	active = new_state
+
+	if(!affected)
+		return
+
+	if(active)
+		activate_with_user()
+	else
+		deactivate_with_user()
+
+///Signal handler for equipping the shield to a slot
+/datum/component/stun_mitigation/proc/shield_equipped(datum/source, mob/living/user, slot)
+	SIGNAL_HANDLER
+	if(!(slot_flags & slot))
+		shield_detatch_from_user()
+		return
+	shield_affect_user(user)
+
+///Signal handler for dropping the shield
+/datum/component/stun_mitigation/proc/shield_dropped(datum/source, mob/user)
+	SIGNAL_HANDLER
+	shield_detatch_from_user()
+
+///Handles the shield setting up for a user, and activating if applicable
+/datum/component/stun_mitigation/proc/shield_affect_user(mob/living/user)
+	if(affected)
+		if(affected == user)
+			return //Already active
+		shield_detatch_from_user() //changing affected
+	affected = user
+	if(active)
+		activate_with_user()
+
+///Actually activates the mitigation effect
+/datum/component/stun_mitigation/proc/activate_with_user()
+	RegisterSignal(affected, COMSIG_LIVING_PROJECTILE_STUN, PROC_REF(on_attack_stun_mitigation))
+
+///Actually deactivates the mitigation effect
+/datum/component/stun_mitigation/proc/deactivate_with_user()
+	UnregisterSignal(affected, COMSIG_LIVING_PROJECTILE_STUN)
+
+///Handles removing the mitigation from a user
+/datum/component/stun_mitigation/proc/shield_detatch_from_user()
+	if(!affected)
+		return
+	SEND_SIGNAL(affected, COMSIG_MOB_SHIELD_DETATCH)
+	deactivate_with_user()
+	affected = null
+
+///attempts to convert incoming hard stuns to soft stuns
+/datum/component/stun_mitigation/proc/on_attack_stun_mitigation(datum/source, list/incoming_stuns, damage_type, penetration)
+	SIGNAL_HANDLER
+
+	var/max_hardstun = max(incoming_stuns[1], incoming_stuns[2]) //stun and weaken
+	if(!max_hardstun)
+		return FALSE
+
+	var/obj/item/parent_item = parent
+	var/mitigation_prob = cover.getRating(damage_type) * (100 - penetration) * 0.01 //pen reduction is a % instead of flat like armor
+	var/status_cover_modifier = 1
+
+	if(mitigation_prob <= 0)
+		return FALSE
+
+	if(parent_item.obj_integrity <= parent_item.integrity_failure)
+		return FALSE
+
+	if(affected.IsSleeping() || affected.IsUnconscious() || affected.IsAdminSleeping())
+		return FALSE
+
+	if(affected.IsStun() || affected.IsKnockdown() || affected.IsParalyzed())
+		status_cover_modifier *= 0.5
+
+	if(iscarbon(affected))
+		var/mob/living/carbon/C = affected
+		if(C.stagger)
+			status_cover_modifier *= 0.4
+
+	mitigation_prob *= status_cover_modifier
```

Comment:
Why set up status_cover_mod at all instead of just multiplying mit prob?

Reply:
Good catch, I had some other interactions but decided to just keep it simple and it was left over.